src/Microsoft.Azure.WebJobs.Host/Queues/Listeners/QueueListener.cs

```diff
@@ -142,15 +142,15 @@ public void Dispose()
                 _stopWaitingTaskSource = new TaskCompletionSource<object>();
             }
 
-            if (!await _queue.ExistsAsync(cancellationToken))
-            {
-                // Back off when no message is available.
-                return CreateBackoffResult();
-            }
-
             IEnumerable<IStorageQueueMessage> batch;
             try
             {
+                if (!await _queue.ExistsAsync(cancellationToken))
```

Comment:
Actually looking at this code more, I don't see why we even have the overhead of a secondary existence check here. We have handling below for "queue not found" so I think we can just rely on that handling (optimistic) and save a network round trip.